### DISCUSSION TITLE
Can we add the requirement for left mouse button click when moving windows, like HyperDock?

### DISCUSSION THREAD

**Discussion by l-qing** [rxhanson/RectanglePro-Community#262] [General]
The current behavior is that the window will move when the shortcut key is pressed and the mouse is moved over it (without the need to click the left mouse button).
![image](https://user-images.githubusercontent.com/9499086/233998899-f8f2b13d-695f-4d22-81a7-ebe335ba4b2a.png)

**Comment by l-qing** [comment]
The function of adjusting the window below is also similar.
Sometimes I accidentally press the modifier key and moving the mouse triggers this behavior.
If you press the modifier key and move the mouse at the same time, can reduce the chance of accidental triggering.

**Comment by rxhanson** [comment]
I have a backlog item for allowing other mouse buttons to trigger move/resize, so I'll see if it makes sense to add it in alongside that feature at that time.

**Reply by l-qing** [reply]
Sure, looking forward to your upcoming versions!

**Reply by derek-shnosh** [reply]
@rxhanson - apologies for the _necro_ here, but curious if this is still under consideration. I'm looking to achieve the following ...

<kbd>CMD</kbd>+<kbd>LEFT-CLICK</kbd> to move the window
<kbd>CMD</kbd>+<kbd>RIGHT-CLICK</kbd> to resize the window

**Reply by rxhanson** [reply]
Thanks for asking.

This is in the app, but it didn't make it into the UI yet.

You can try it out with the following defaults Terminal commands:

```
defaults write com.knollsoft.Hookshot clickMoveMouseButton -int 1
defaults write com.knollsoft.Hookshot clickResizeMouseButton -int 2
```

This sets the move mouse button 1 (left click), and the resize to mouse button 2 (right click).

```
defaults write com.knollsoft.Hookshot clickMoveResizeUsesModifiers -int 1
```
This makes it so the modifiers you have configured need to be applied in order to trigger it, so you'd want to set those to command in the UI.

Restart the app after.

Let me know if you get a chance to try it out.

**Reply by derek-shnosh** [reply]
Thanks for the fast response! Since we're using <kbd>LEFT-CLICK</kbd> for **Move** and <kbd>RIGHT-CLICK</kbd> for **Resize** (in this scenario), is it somehow possible to allow <kbd>CMD</kbd> to be used as the Modifier for both?

<img width="842" height="524" alt="image" src="https://github.com/user-attachments/assets/9f9a51b6-0cad-40e8-83a5-de298a567e41" />

Also, I know this one may be a bit of a stretch given how Mac differs from any other OS in terms of keyboard (function key) layout/assignment ... but is it possible to exclude the <kbd>CMD</kbd> modifier from interfering with <kbd>CMD</kbd>-dragging Menu Bar items (to reorganize them?)

**Reply by rxhanson** [reply]
Thanks for the quick turnaround. In the upcoming release I'll remove that safeguard when this setting is in place and make sure that they are both handled properly. It's possible to set this by manually editing the config, either by defaults or config export/edit/import, but it's fairly tedious and I haven't tested them both having the same modifier. If you are really wanting to try it out before that restriction is removed, you'd be editing the json for the "gestures" key, and setting the flags for 33 (resize) to be the same as 32 (move) - if it were me I'd probably just wait. I'm hoping to have another release out soon.

**Reply by derek-shnosh** [reply]
Excellent!

I currently have <kbd>CMD</kbd> for **Move** and <kbd>CMD</kbd>+<kbd>OPT</kbd> for **Resize** ...

```sh
~ $ defaults read com.knollsoft.Hookshot gestures
[32,{"trail":0,"flags":1048576},33,{"trail":0,"flags":1572864}]
~ $
```

I adjusted with ...

```sh
defaults write com.knollsoft.Hookshot gestures -string '[32,{"trail":0,"flags":1048576},33,{"trail":0,"flags":1048576}]'
```

And it's now working as desired after relaunching the app, and is even reflected in the UI.

<img width="710" height="508" alt="image" src="https://github.com/user-attachments/assets/ca1587c9-b975-4795-890b-07a8f332957b" />

I greatly appreciate the response and insight (as a new licensee 👌)!

**Reply by derek-shnosh** [reply]
@rxhanson - now that I have the <kbd>CMD</kbd> modifier to work with left and right click to move and resize windows (respectively) ... this seems to be interfering with <kbd>CMD</kbd>+dragging items in the Menu Bar - is there any way to remedy this?

**Reply by rxhanson** [reply]
Thanks for the heads up. I'll see what I can do.
